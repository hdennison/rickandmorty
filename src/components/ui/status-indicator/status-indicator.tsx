import { CircleIcon, IconProps } from "../icon/icon";

const statusColors = {
  Alive: "green",
  Dead: "red",
  unknown: "grey",
};

type Props = IconProps & {
  status: keyof typeof statusColors;
};
export const StatusIndicator = ({ status, ...props }: Props) => (
  <CircleIcon
    color={statusColors[status]}
    fill={statusColors[status]}
    {...props}
  />
);
