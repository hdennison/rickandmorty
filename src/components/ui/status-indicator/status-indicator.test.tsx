import { describe, expect, test } from "vitest";
import { render } from "vitest-browser-react";
import { StatusIndicator } from "./status-indicator";

const allStatus = ["Alive", "Dead", "unknown"] as const;

describe("variants", () => {
  allStatus.forEach((status) => {
    test(`Status: ${status}`, () => {
      const indicator = render(<StatusIndicator status={status} />);

      // Snapshots are not the best solution, it woould be much better to set up Storybook + Chromatic
      expect(indicator).toMatchSnapshot();
    });
  });
});
