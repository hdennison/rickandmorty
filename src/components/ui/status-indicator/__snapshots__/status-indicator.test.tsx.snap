// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`variants > Status: Alive 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    
    
    
  


    <div>
      <svg
        class="lucide lucide-circle shrink-0"
        fill="green"
        height="24"
        stroke="green"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="lucide lucide-circle shrink-0"
      fill="green"
      height="24"
      stroke="green"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
    </svg>
  </div>,
  "debug": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholder": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`variants > Status: Dead 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    
    
    
  


    <div>
      <svg
        class="lucide lucide-circle shrink-0"
        fill="red"
        height="24"
        stroke="red"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="lucide lucide-circle shrink-0"
      fill="red"
      height="24"
      stroke="red"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
    </svg>
  </div>,
  "debug": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholder": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`variants > Status: unknown 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    
    
    
  


    <div>
      <svg
        class="lucide lucide-circle shrink-0"
        fill="grey"
        height="24"
        stroke="grey"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="lucide lucide-circle shrink-0"
      fill="grey"
      height="24"
      stroke="grey"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
    </svg>
  </div>,
  "debug": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholder": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
